service/src/main/java/tds/session/repositories/ExternRepository.java

```diff
@@ -0,0 +1,14 @@
+package tds.session.repositories;
+
+import java.util.Optional;
+
+import tds.session.Extern;
+
+public interface ExternRepository {
+    /**
+     * Fetches the extern by given client name
+     * @param clientName client name for extern
+     * @return optional containing {@link tds.session.Extern Extern} otherwise empty Optional
+     */
+    Optional<Extern> getExternByClientName(String clientName);
```

Comment:
I was thinking Exam/Legacy Student will need to handle when there is insufficient data to complete the task.  For example, a new exam cannot be created without the extern configured properly so exam is going to return an error in that case.  Not sure it'll be an Exception but it'll be an error.

If we do decide to throw here what would be the HTTP return code?  500?  The controller currently returns a 404.